[Doc] Add deployment documentation about CeleryExecutor (#105)
flink-extended/ai-flow: docs/content/deployment/deploying_scheduler.md

### CeleryExecutor

#### Standalone Deployment

> 1. Set up Celery broker and result backend. CeleryExecutor take advantage of the [Celery](https://docs.celeryproject.org/en/stable/index.html) to 

Reviewer: ```suggestion
1. Set up Celery broker and result backend. `CeleryExecutor` takes advantage of the [Celery](https://docs.celeryproject.org/en/stable/index.html) to 
```

>    scale out the number of workers. For this to work, you need to setup a broker and a result backend (RabbitMQ, Redis, ...). 

Reviewer: It's recommended to tell the users how to setup a broker and a result backend, which could give a link to users.

>    scale out the number of workers. For this to work, you need to setup a broker and a result backend (RabbitMQ, Redis, ...). 

Reviewer: ```suggestion
   scale out the number of workers. For this to work, you need to set up a broker and a result backend (RabbitMQ, Redis, ...). 
```

> 2. Change the Airflow configuration. You need to change your airflow.cfg to point the executor parameter to CeleryExecutor and provide the neccess related Celery settings.

Reviewer: ```suggestion
2. Change the Airflow configuration. You need to change your airflow.cfg to set the `executor` parameter to `CeleryExecutor` and provide the neccess related Celery settings.
```

3. Set up and kick off all Celery workers. Here are some imperative requirements for your workers:
   * Airflow needs to be installed, and the CLI needs to be in the path.
   * Airflow Configuration settings should be homogeneous across the cluster.

   After setting up Celery workers, you can kick off a worker with following command.
   ```
   airflow celery worker
   ```
   To stop a worker running on a machine you can use:
   ```
   airflow celery stop
   ```
   
4. Start the Scheduler
    ```
    start-airflow.sh
    ```

#### Deploy with Docker Compose

> Deploying CeleryExecutor is a bit complicated, you can follow this guide to quickly start AIFlow and Scheduler with CeleryExecutor within Docker.

Reviewer: `Deploying CeleryExecutor is a bit complicated` could be removed.

> Note: the Docker Compose below may be not enough to run production-ready Docker Compose AIFlow installation using it.

Reviewer: Where is the Docker Compose below?

This is truly quick-start docker-compose for you to get your hands dirty with AIFlow and CeleryExecutor.

##### Before you begin

Make sure you have Docker and Docker Compose installed on your workstation,
otherwise please follow these steps to install the tools.

> 1. Install [Docker Community Edition (CE)](https://docs.docker.com/engine/install/) on your workstation. Depending on the OS, you may need to configure your Docker instance to use 4.00 GB of memory for all containers to run properly.

Reviewer: ```suggestion
1. Install [Docker Community Edition (CE)](https://docs.docker.com/engine/install/) on your workstation. Depending on the OS, you may need to configure your Docker instance to use at leaset 4GB of memory for all containers to run properly.
```

2. Install [Docker Compose](https://docs.docker.com/compose/install/) on your workstation.

> Run following commands to check if the tools are installed successfully.

Reviewer: ```suggestion
Run the following commands to check whether the tools are installed successfully:
```

```shell script
docker --version
docker-compose --version
```

##### docker-compose.yaml
To deploy AIFlow with CeleryExecutor on Docker Compose, you should fetch [docker-compose.yaml](http://raw.githubusercontent.com/flink-extended/ai-flow/master/docker-compose.yaml).

```shell script
curl -LfO http://raw.githubusercontent.com/flink-extended/ai-flow/master/docker-compose.yaml
```

This file contains several service definitions:

* AI Flow servers - AI Flow Server, Notification Server and Scheduler deployed in one container.
* Workers - The workers who execute the tasks given by the Scheduler.
* Redis - The broker that forwards messages from Scheduler to worker.
* Flower -  The [flower app](https://flower.readthedocs.io/en/latest/) for monitoring the environment. It is available at http://localhost:5555.
* MySQL - The database. Only MySQL is supported in distributed mode for now.
* HDFS - The hadoop dfs which is used to store the source code or any resources of your projects.

##### Initializing Environment

> This command should be executed in the directory that contains docker-compose.yaml you just downloaded.

Reviewer: Should this first explain that the operating environment can be initialized by using the following command, and then mention where to execute this command?

Author: Agree

> It will help start up database, redis, hdfs and 2 workers and also initialize execution environment of all servers.

Reviewer: It's recommended to tell the users how to check whether the environment has been initialized.

```shell script
docker-compose up -d --scale airflow-worker=2
```

##### Starting Services

Execute following command to start up all services.
```shell script
docker exec ai-flow_aiflow-server_1 bash -c "start-all-aiflow-services.sh"
```

> Once all servers started, you can visit the [Flower web](http://127.0.0.1:5555/) to view the celery workers.

Reviewer: It's recommended to add a picture to show the celery workers in WebUI.

* Flower -  The [flower app](https://flower.readthedocs.io/en/latest/) for monitoring the environment. It is available at http://localhost:5555.
* MySQL - The database. Only MySQL is supported in distributed mode for now.
* HDFS - The hadoop dfs which is used to store the source code or any resources of your projects.

##### Initializing Environment

This command will help start up database, redis, hdfs and 3 workers and also initialize execution environment of all servers.
It should be executed in the directory that contains docker-compose.yaml you just downloaded.
```shell script
docker-compose up -d --scale airflow-worker=3
```
You can run ```docker ps ``` to check whether all components are started and healthy.

##### Starting Services

Execute following command to start up all services.
```shell script
docker exec ai-flow_aiflow-server_1 bash -c "start-all-aiflow-services.sh"
```
Once all servers started, you can visit the [Flower web](http://127.0.0.1:5555/) to view the celery workers.

##### Running Example
```shell script
docker exec ai-flow_aiflow-server_1 bash -c "python /opt/aiflow/examples/celery/workflows/simple_workflow/simple_workflow.py"
```

(configuration)=

> ## Configuration

Reviewer 2: This section shows a list of all the configurations we added to Airflow to make it works with AIFlow.
->
This section shows a list of configurations you may need to change to Airflow to make it works with AIFlow.
